src/Expander.hs

```diff
@@ -115,12 +115,13 @@ expandModule thisMod src = do
   local (set (expanderLocal . expanderModuleName) thisMod) do
     lang <- mustBeModName (view moduleLanguage src)
     initializeLanguage lang
-    modBodyDest <- liftIO $ newModBodyPtr
-    modifyState $ set expanderModuleTop $ Just modBodyDest
+    declTreeDest <- liftIO $ newDeclTreePtr
+    vp <- newDeclValidityPtr
```

Comment:
This DeclValidityPtr gets filled, but isn't read anywhere. An alternative would be to use `Maybe DeclValidityPtr` everywhere, and to only fill it when it's a `Just`.